packages/app/components/Settings/CourseRoster.tsx

```diff
@@ -0,0 +1,122 @@
+import { SearchOutlined } from "@ant-design/icons";
+import { API } from "@koh/api-client";
+import { Role } from "@koh/common";
+import { Divider, Input, List, Pagination, Spin } from "antd";
+import Avatar from "antd/lib/avatar/avatar";
+import { useState } from "react";
+import { ReactElement } from "react";
+import styled from "styled-components";
+import useSWR from "swr";
+
+type CourseRosterProps = { courseId: number };
+
+type RenderTableProps = {
+  courseId: number;
+  role: Role;
+  listTitle: string;
+  displaySearchBar: boolean;
+  searchPlaceholder: string;
+};
+
+const CourseRosterComponent = styled.div`
+  margin-left: auto;
+  margin-right: auto;
+`;
+
+export default function CourseRoster({
+  courseId,
+}: CourseRosterProps): ReactElement {
+  return (
+    <CourseRosterComponent>
+      <h1>Course Roster</h1>
+      <RenderTable
+        courseId={courseId}
+        role={Role.PROFESSOR}
+        listTitle={"Professors"}
+        displaySearchBar={false}
+        searchPlaceholder=""
+      />
+      <br />
+      <RenderTable
+        courseId={courseId}
+        role={Role.TA}
+        listTitle={"Teaching Assistants"}
+        displaySearchBar={true}
+        searchPlaceholder="Search TAs"
+      />
+      <br />
+      <RenderTable
+        courseId={courseId}
+        role={Role.STUDENT}
+        listTitle={"Students"}
+        displaySearchBar={true}
+        searchPlaceholder="Search students"
+      />
+      <br />
+      <Divider />
+    </CourseRosterComponent>
+  );
+}
+
+function RenderTable({
+  courseId,
+  role,
+  listTitle,
+  displaySearchBar,
+  searchPlaceholder,
+}: RenderTableProps): ReactElement {
+  const [page, setPage] = useState(1);
+  const [search, setSearch] = useState("");
+  const handleSearch = (event) => {
```

Comment:
it looks like this is submitting an API request with every character you type, which based on my manual testing feels a bit rough - you have to type a character, wait for it to load, then type another character. I think we can probably make the UX a better by either debouncing the request (basically, set a timeout and only submit the request if x seconds pass without the user typing something), or just only submit the search request when the user hits the enter key. Hitting the enter key is probably the easier approach, but if you wanna do the debounce feel free to talk to me if you want help or anything :P

Reply:
oh, and I also noticed that it seems like you can search by people's names, but not by emails 🤔 I don't think it's too important, but if it's not difficult to make that change it might be nice to have as well

Reply:
Good point, I took the enter approach as it seemed intuitive to me and made the change

Think I would have to make some changes to the backend to allow for email searching, will do that real quick